### DISCUSSION TITLE
被动联动计算与预期不符合

### DISCUSSION THREAD

**Discussion by febugcoder** [alibaba/formily#3832] [Q&A 2.x] [answered]
# 需求
有A、B、C、D四个字段，其中C = A / B，D = C * B
[https://codesandbox.io/s/admiring-glade-puqoq5?file=/src/App.js:539-551](https://codesandbox.io/s/admiring-glade-puqoq5?file=/src/App.js:539-551)
# 操作步骤
1. 输入A = 2
2. 输入B = 1
3. 此时C会计算出来2，但是D没有算出值
# 问题
为什么D会计算不出来结果？当C被计算出来之后，为什么D的onFieldReact没有再跑一次？

**Comment by yanhao98** [comment]
<img width="664" alt="image" src="https://github.com/alibaba/formily/assets/37316281/60df28d4-5b0b-445f-9e92-af7dbbba41c3">


<img width="378" alt="image" src="https://github.com/alibaba/formily/assets/37316281/4f421bd9-1dd6-496c-8351-66c56c0ba583">

我在Vue试了一下，是正常的，不太懂React。提供了一下对比日志。

**Reply by febugcoder** [reply]
感谢尝试！
现在的情况应该是像图2一样，看图2：
1. B设置了1之后，先执行了onFieldReact D，这时候C还是undefined
2. 然后紧接着执行onFieldReact C，这时候C的值会变化
但你看你的日志，onFieldReact D并没有再继续执行了，所以计算不出结果

**Reply by febugcoder** [reply]
@janryWang 大佬如果看到了是否可以解答一下呢

**Comment by janryWang** [comment, marked as answer]
提个 bug ？这个的确有点问题

**Reply by febugcoder** [reply]
感谢，已提https://github.com/alibaba/formily/issues/3837
